src/coreclr/tools/SuperFileCheck/Program.cs

```diff
@@ -0,0 +1,623 @@
+            foreach (var prefix in checkPrefixes)
+            {
+                Console.Error.WriteLine($"    {prefix}");
+            }
+            Console.ResetColor();
+        }
+
+        static void PrintErrorNoInputFileFound()
+        {
+            Console.ForegroundColor = ConsoleColor.Red;
+            Console.Error.WriteLine($"{CommandLineInputFile} is required.");
+            Console.ResetColor();
+        }
+
+        static void PrintHelp()
+        {
+            Console.Write(Environment.NewLine);
+            Console.WriteLine("USAGE: SuperFileCheck [options] <check-file>");
+            Console.WriteLine("USAGE: SuperFileCheck <super-option> <check-file> [options]");
+            Console.Write(Environment.NewLine);
+            Console.WriteLine("SUPER OPTIONS:");
+            Console.Write(Environment.NewLine);
+            Console.WriteLine($"  --csharp                       - An {CommandLineInputFile} is required.");
+            Console.WriteLine($"                                   <check-file> must be a C# source file.");
+            Console.WriteLine($"                                   Methods must not have duplicate names.");
+            Console.WriteLine($"                                   Methods must be marked as not inlining.");
+            Console.WriteLine($"                                   One or more methods are required.");
+            Console.WriteLine($"                                   Prefixes are determined by {CommandLineCheckPrefixes}.");
+            Console.WriteLine($"  --csharp-list-method-names     - Print a space-delimited list of method names to be");
+            Console.WriteLine($"                                   supplied to environment variable DOTNET_JitDisasm.");
+            Console.WriteLine($"                                   <check-file> must be a C# source file.");
+            Console.WriteLine($"                                   Methods must not have duplicate names.");
+            Console.WriteLine($"                                   Methods must be marked as not inlining.");
+            Console.WriteLine($"                                   Prints nothing if no methods are found.");
+            Console.WriteLine($"                                   Prefixes are determined by {CommandLineCheckPrefixes}.");
+        }
+
+        static string? TryFindDuplicateMethodName(MethodDeclarationInfo[] methodDeclInfos)
+        {
+            var set = new HashSet<string>();
+
+            var duplicateMethodDeclInfo =
+                methodDeclInfos.FirstOrDefault(x =>
+                {
+                    return !set.Add(x.Name);
+                });
+
+            if (duplicateMethodDeclInfo.Name != null)
+            {
+                return duplicateMethodDeclInfo.Name;
+            }
+            else
+            {
+                return null;
+            }
```

Comment:
Just `return duplicateMethodDeclInfo.Name;`?

Reply:
Or do you want `return duplicateMethodDeclInfo?.Name;` (or no `.Name` in the `if` if you keep that form)?

Reply:
`return duplicateMethodDeclInfo.Name;` will work since `return duplicateMethodDeclInfo` is a struct and default initialization will make `.Name` a `null`.